### DISCUSSION TITLE
How to export SBOM CRD to CycloneDX compatible JSON

### DISCUSSION THREAD

**Discussion by georok** [aquasecurity/trivy-operator#1936] [Help & Support]
I updated the trivy operator to 0.19.1 and it successfully generates SBOM CRDs. I can export those CRDs with 
"kubectl get sbomreports/"name" -n "namespace" -o json > sbom_report.json
but the exported json is not CycloneDX compatible. What am I doing wrong and how should I edit the exported JSON to make it CycloneDX compatible?
thanks

**Comment by darren128** [comment]
Is there any effort to make these CRDs contain a CycloneDX formatted report or any other format that trivy cli supports, ie:
```
$ trivy image --format cyclonedx --output result.json alpine:3.15
```

**Comment by Kankarollo** [comment]
Cyclonedx SBOM is inside the kubernetes CRDs in .reports.components.*. 
You can extract it with 
```
kubectl get sbomreports/"name" -n "namespace" -o json | jq '.report.components' > sbom_report.json
```
Keep in mind that the new versions of Trivy operator (at least for me it was 0.23.0) it supports already cyclonedx 1.6 scheme so old validators will mark it as a not valid. You can ofcourse just change "specVersion": "1.6", to "specVersion": "1.5".
